Tests/DistributedClusterTests/Plugins/ClusterSingleton/ClusterSingletonPluginTests.swift

```diff
@@ -59,6 +59,48 @@ final class ClusterSingletonPluginTests: SingleClusterSystemXCTestCase {
         // if this were true we would have crashed by a duplicate name already, but let's make sure:
         singletonID.shouldNotEqual(greeterID)
     }
+    
+    func test_plugin_hooks() async throws {
+        let actorId = "actorHookId"
+        let hookFulfillment = self.expectation(description: "actor-hook")
+        let plugin = TestClusterHookPlugin { actor in
+            /// There are multiple internal actors fired, we only checking for `ActorWithId`
+            guard let actor = actor as? ActorWithId else { return }
+            let id = try? await actor.getId()
+            XCTAssertEqual(id, actorId, "Expected \(actorId) as an id")
+            hookFulfillment.fulfill()
+        }
+        let testNode = await setUpNode("test-hook") { settings in
+            settings.enabled = false
+            settings += plugin
+        }
+
+        let _ = ActorWithId(actorSystem: testNode, id: actorId)
+        await fulfillment(of: [hookFulfillment])
+    }
+    
+    actor TestClusterHookPlugin: _Plugin, PluginActorLifecycleHook {
```

Comment:
Oh plugin was hidden still huh. About time to make it public -- maybe as we cleanup the how we use them with your work here!

Reply:
🫡
Will remove `_` from Plugin